src/eos-crash-metrics.c

```diff
@@ -0,0 +1,185 @@
+}
+
+static OstreeSysroot *
+load_ostree_sysroot (void)
+{
+  OstreeSysroot *sysroot = ostree_sysroot_new_default ();
+  g_autoptr(GError) error = NULL;
+
+  if (!ostree_sysroot_load (sysroot, NULL, &error))
+    {
+      g_warning ("Unable to get current OSTree sysroot: %s", error->message);
+      return NULL;
+    }
+
+  return sysroot;
+}
+
+static char *
+get_eos_ostree_repo_url (OstreeRepo *repo)
+{
+  GKeyFile *config = NULL;
+  g_autoptr(GError) error = NULL;
+  char *url = NULL;
+
+  config = ostree_repo_get_config (repo);
+
+  if (!(url = g_key_file_get_string (config, "remote \"eos\"", "url", &error)))
+    {
+      g_warning ("Unable to read OSTree config for eos remote URL: %s", error->message);
+      return NULL;
+    }
+
+  return url;
+}
+
+static char *
+get_eos_ostree_deployment_commit (OstreeSysroot *sysroot, OstreeRepo *repo)
+{
+  OstreeDeployment *deployment = ostree_sysroot_get_booted_deployment (sysroot);
+  GKeyFile *config = NULL;
+  g_autoptr(GError) error = NULL;
+  g_autofree char *refspec = NULL;
+  char *commit = NULL;
+
+  if (!deployment)
+    {
+      g_warning ("OSTree deployment is not currently booted, cannot read state");
+      return NULL;
+    }
+
+  config = ostree_deployment_get_origin (deployment);
+
+  if (!(refspec = g_key_file_get_string (config, "origin", "refspec", &error)))
+    {
+      g_warning ("Unable to read OSTree refspec for booted deployment: %s", error->message);
+      return NULL;
+    }
+
+  if (!ostree_repo_resolve_rev (repo, refspec, FALSE, &commit, &error))
+    {
+      g_warning ("Unable to resolve revision for refpsec %s: %s", refspec, error->message);
+      return NULL;
+    }
+
+  return commit;
+}
+
+/* This is the way that the kernel gives us paths ... */
+static const char *blacklisted_prefixes[] =
+{
+  "!home",
+  "!sysroot!home"
+};
+
+static gboolean
+is_blacklisted_path (const char *path)
+{
+  const size_t n_blacklisted_prefixes = G_N_ELEMENTS (blacklisted_prefixes);
+  size_t i = 0;
+
+  for (; i < n_blacklisted_prefixes; ++i)
+    {
+      if (g_str_has_prefix (path, blacklisted_prefixes[i]))
+        return TRUE;
+    }
+
+  return FALSE;
+}
+
+int
+main (int argc, char **argv)
+{
+  const gchar *path = NULL;
+  gint16 signal = 0;
+  gint64 timestamp = 0;
+  g_autoptr(OstreeSysroot) sysroot = NULL;
+  g_autoptr(OstreeRepo) repo = NULL;
+  g_autoptr(GError) error = NULL;
+  g_autofree char *url = NULL;
+  g_autofree char *commit = NULL;
+
+  if (argc != EXPECTED_NUMBER_ARGS + 1)
+    {
+      g_warning("Not enough arguments passed (%i, expected %i).", argc - 1, EXPECTED_NUMBER_ARGS);
+      return EXIT_FAILURE;
+    }
+
+  path = argv[1];
+  signal = atoi (argv[2]);
+  timestamp = atoll (argv[3]);
+
+  if (is_blacklisted_path (path))
+    {
+      g_message ("%s is blacklisted, not reporting crash", path);
+      return EXIT_SUCCESS;
+    }
+
+  sysroot = load_ostree_sysroot ();
+
+  if (ostree_sysroot_get_repo (sysroot, &repo, NULL, &error))
```

Comment:
Nitpick: prefer inverting this check and returning early when it fails, so that you can remove the `else` block.